startup/21-cameras.py

```diff
@@ -0,0 +1,107 @@
+hfvlm_cam_tiff = EpicsSignal('XF:05IDD-BI:1{Mscp:1-Cam:1}TIFF1:Capture',
+                        write_pv='XF:05IDD-BI:1{Mscp:1-Cam:1}TIFF1:Capture',
+                        name='hfvlm_cam_tiff')
+
+
+### BPM Camera
+class SRXTIFFPlugin(TIFFPlugin, FileStoreTIFF,
+                    FileStoreIterativeWrite):
+    file_number_sync = None
+
+
+class BPMCam(SingleTrigger, AreaDetector):
+    cam = C(AreaDetectorCam, '')
+    image_plugin = C(ImagePlugin, 'image1:')
+
+    # tiff = C(SRXTIFFPlugin, 'TIFF1:',
+    #          #write_path_template='/epicsdata/bpm1-cam1/2016/2/24/')
+    #          #write_path_template='/epicsdata/bpm1-cam1/%Y/%m/%d/',
+    #          #root='/epicsdata', reg=db.reg)
+    #          write_path_template='/nsls2/xf05id1/data/bpm1-cam1/%Y/%m/%d/',
+    #          root='/nsls2/xf05id1')
+    roi1 = C(ROIPlugin, 'ROI1:')
+    roi2 = C(ROIPlugin, 'ROI2:')
+    roi3 = C(ROIPlugin, 'ROI3:')
+    roi4 = C(ROIPlugin, 'ROI4:')
+    stats1 = C(StatsPlugin, 'Stats1:')
+    stats2 = C(StatsPlugin, 'Stats2:')
+    stats3 = C(StatsPlugin, 'Stats3:')
+    stats4 = C(StatsPlugin, 'Stats4:')
+    # this is flakey?
+    # stats5 = C(StatsPlugin, 'Stats5:')
+    # pass  # is this needed?
+
+
+bpmAD = BPMCam('XF:05IDA-BI:1{BPM:1-Cam:1}', name='bpmAD', read_attrs=[])
+bpmAD.read_attrs = ['stats1', 'stats2', 'stats3', 'stats4']
+bpmAD.stats1.read_attrs = ['total']
+bpmAD.stats2.read_attrs = ['total']
+bpmAD.stats3.read_attrs = ['total']
+bpmAD.stats4.read_attrs = ['total']
+
+
+### HF VLM
+# Does this belong here or in microES?
+class SRXHFVLMCam(SingleTrigger, AreaDetector):
+    cam = C(AreaDetectorCam, 'cam1:')
+    image_plugin = C(ImagePlugin, 'image1:')
+    proc1 = C(ProcessPlugin, 'Proc1:')
+    stats1 = C(StatsPlugin, 'Stats1:')
+    stats2 = C(StatsPlugin, 'Stats2:')
+    stats3 = C(StatsPlugin, 'Stats3:')
+    stats4 = C(StatsPlugin, 'Stats4:')
+    roi1 = C(ROIPlugin, 'ROI1:')
+    roi2 = C(ROIPlugin, 'ROI2:')
+    roi3 = C(ROIPlugin, 'ROI3:')
+    roi4 = C(ROIPlugin, 'ROI4:')
+    over1 = C(OverlayPlugin, 'Over1:')
+    trans1 = C(TransformPlugin, 'Trans1:')
+    tiff = C(SRXTIFFPlugin, 'TIFF1:',
+             write_path_template='/epicsdata/hfvlm/%Y/%m/%d/',
+             root='/epicsdata')
+             # write_path_template='/nsls2/xf05id1/data/hfvlm/%Y/%m/%d/',
+             # root='/nsls2/xf05id1')
+
+try:
+    hfvlmAD = SRXHFVLMCam('XF:05IDD-BI:1{Mscp:1-Cam:1}', name='hfvlm', read_attrs=['tiff'])
+    hfvlmAD.read_attrs = ['tiff', 'stats1', 'stats2', 'stats3', 'stats4']
+    hfvlmAD.tiff.read_attrs = []
+    hfvlmAD.stats1.read_attrs = ['total']
+    hfvlmAD.stats2.read_attrs = ['total']
+    hfvlmAD.stats3.read_attrs = ['total']
+    hfvlmAD.stats4.read_attrs = ['total']
+except TimeoutError:
+    hfvlmAD = None
+    print('\nCannot connect to HF VLM Camera. Continuing without device.\n')
+except:
+    hfvlmAD = None
+    print('\nUnexpected error connecting to HF VLM Camera.\n', sys.exc_info()[0], end='\n\n')
```

Comment:
I'd recommend to `import sys` on top of this file, so it can work after it's copied to another beamline, for example.

Reply:
Imported sys.
But also changed this to "except Exception as ex" and printed that exception.